### PR TITLE
New

### PR DESCRIPTION
**Pull request by VadimDrobyazko**
[DEMO LINK](https://vadimdrobyazko.github.io/react_todo-app-with-api/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,206 @@
...
+  const [isDisableInput, setIsDisableInput] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const completedTodos = todos.every((todo) => todo.completed);
+
+  const fetchData = async () => {
+    const todosFromServer = await getTodo(USER_ID);
+
+    try {
+      setTodo(todosFromServer);
+      setIsLoading(true);
```
Comment (IvanFesenko) [inline src/App.tsx:33]: enabling loader should be before request

File: src/App.tsx [hunk 2]
```diff
@@ -1,24 +1,206 @@
...
+        case FilterType.ACTIVE:
+          return !todo.completed;
+
+        case FilterType.COMPLETED:
+          return todo.completed;
+
+        default:
+          return todo;
+      }
+    });
```
Comment (IvanFesenko) [inline src/App.tsx:57]: No need to filter if FilterType.ALL
Also filter callbcack shoud return boolean you have todo instead

File: src/App.tsx [hunk 3]
```diff
@@ -1,24 +1,206 @@
...
+  }, [todos, filterType]);
+
+  const handleAddTodo = async (title: string) => {
+    try {
+      setIsDisableInput(true);
+
+      const data = {
+        id: Math.floor(Math.random() * 100),
```
Comment (IvanFesenko) [inline src/App.tsx:65]: Use crypto.randomuuid()

File: src/App.tsx [hunk 4]
```diff
@@ -1,24 +1,206 @@
...
+        title,
+        completed: false,
+      };
+
+      const newTodo = await postTodo(data);
+
+      setTodo(todo => [...todo, newTodo]);
+      setTempTodo({ ...data, id: 0 });
+    } catch {
+      setError('Unable to add a todo');
```
Comment (IvanFesenko) [inline src/App.tsx:76]: No error visible

File: src/App.tsx [hunk 5]
```diff
@@ -1,24 +1,206 @@
...
+  };
+
+  useEffect(() => {
+    if (error) {
+      const timeout = setTimeout(() => {
+        setError('');
+      }, 3000);
+
+      clearTimeout(timeout);
+    }
```
Comment (IvanFesenko) [inline src/App.tsx:146]: Fix same as on previous task

File: src/components/AddInput/AddInput.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  const [input, setInput] = useState('');
+  const inputElem = useRef<HTMLInputElement>(null);
+
+  const handleFormSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    try {
+      e.preventDefault();
+      const title = input.trim();
+
+      if (title) {
+        await handleAddTodo(title);
```
Comment (IvanFesenko) [inline src/components/AddInput/AddInput.tsx:28]: async is redudant here

File: src/components/TodoFilter/TodoFilter.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+      <nav className="filter">
+        <a
+          href="#/"
+          className={cn(
+            'filter__link',
+            {
+              selected: filterType === FilterType.ALL,
+            },
+          )}
+          onClick={() => setFilterType(FilterType.ALL)}
```
Comment (IvanFesenko) [inline src/components/TodoFilter/TodoFilter.tsx:36]: Create separate handler instead of using anonymous function. Fix all similar cases

**Review by IvanFesenko — CHANGES_REQUESTED** [review]
Same iisue with adding todos

**Review by sTorba24 — APPROVED** [review]
Well done!

**Pull request closed without merging**